SingularityService/src/main/java/com/hubspot/singularity/hooks/LoadBalancerClientImpl.java

```diff
@@ -68,6 +72,54 @@ public LoadBalancerClientImpl(SingularityConfiguration configuration, ObjectMapp
     this.mesosProtosUtils = mesosProtosUtils;
   }
 
+  public String getLoadBalancerUri() {
+    return loadBalancerUri;
+  }
+
+  public Optional<Map<String, String>> getLoadBalancerQueryParams() {
+    return loadBalancerQueryParams;
+  }
+
+  public long getLoadBalancerTimeoutMillis() {
+    return loadBalancerTimeoutMillis;
+  }
+
+  public Optional<String> getTaskLabelForLoadBalancerUpstreamGroup() {
+    return taskLabelForLoadBalancerUpstreamGroup;
+  }
+
+  public MesosProtosUtils getMesosProtosUtils() {
+    return mesosProtosUtils;
+  }
+
+  private String getLoadBalancerStateUri (){
+    return loadBalancerUri.replace("request", "state");
+  }
+
+  private Collection<BaragonServiceState> getBaragonServiceStates () {
+    final String loadBalancerStateUri = getLoadBalancerStateUri();
+    final BoundRequestBuilder requestBuilder = httpClient.prepareGet(loadBalancerStateUri);
+    final Request request = requestBuilder.build();
+    try {
+      ListenableFuture<Response> future = httpClient.executeRequest(request);
+      Response response = future.get(loadBalancerTimeoutMillis, TimeUnit.MILLISECONDS);
+      if (!JavaUtils.isHttpSuccess(response.getStatusCode())) {
+        LOG.info(String.format("Response status code %s", response.getStatusCode()));
+      }
+      return objectMapper.readValue(response.getResponseBodyAsBytes(), new TypeReference<List<BaragonServiceState>>() {});
+    } catch (IOException | InterruptedException | ExecutionException | TimeoutException e) {
+      e.printStackTrace();
+      return Collections.emptyList();
+    }
+  }
+
+  public Collection<UpstreamInfo> getUpstreams () {
```

Comment:
related to my comment in the poller, we can probably make this take an argument of `String requestId` and return the list of upstreams for only that requestId